services/housetables/src/main/java/com/linkedin/openhouse/housetables/services/UserTablesServiceImpl.java

```diff
@@ -257,17 +253,14 @@ private Page<UserTableDto> searchTables(UserTable userTable, int page, int size,
                     userTable.getMetadataLocation(),
                     userTable.getStorageType(),
                     userTable.getCreationTime(),
-                    pageable)
+                    injectDefaultSort(pageable, "tableId"))
                 .map(userTableRow -> userTablesMapper.toUserTableDto(userTableRow)),
         MetricsConstant.HTS_PAGE_SEARCH_TABLES_TIME);
   }
 
-  private Pageable createPageable(int page, int size, String sortBy, String defaultSortBy) {
-    Sort sort =
-        StringUtils.isEmpty(sortBy)
-            ? Sort.by(defaultSortBy).ascending()
-            : Sort.by(sortBy).ascending();
-    return PageRequest.of(page, size, sort);
+  private Pageable injectDefaultSort(Pageable pageable, String defaultSortBy) {
+    Sort sort = pageable.getSortOr(Sort.by(defaultSortBy).ascending());
+    return PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), sort);
```

Comment:
Pageable was constructed in HtsApiHandler and passed over to UserTablesService. Now in this method default sorting is applied and pageable is constructed again. Can we apply soft and construct Pageable only once here in the service layer?